base/task.jl

```diff
@@ -1184,7 +1196,12 @@ checktaskempty = Partr.multiq_check_empty
 @noinline function poptask(W::StickyWorkqueue)
     task = trypoptask(W)
```

Comment:
I think this should fix your concern that we don't do enough of the `jl_task_get_next` loop before switching to the scheduler to spin and wait (though we could also move more of that loop here)
```suggestion
    task = trypoptask(W)
    if !(task isa Task)
        process_events()
        task = trypoptask(W)
    end
```